src/helpers/logDecorator.ts

```diff
@@ -0,0 +1,39 @@
+import showTable from "../methods/showTable.js"
+import { defaultOptions } from "../types.js"
+import log from "./log.js"
+
+export default function logDecorator(simpleData: any, func: Function, ...args: any[]) {
+
+    const start = Date.now()
+
+    const parameters = args
+    let options
+
+    if (
+        args.length > 0 &&
+        typeof args[args.length - 1] === "object" &&
+        Object.keys(defaultOptions).includes(Object.keys(args[args.length - 1])[0])
+    ) {
+        options = { ...simpleData.options, ...args[args.length - 1] }
+        parameters[args.length - 1] = options
+    } else {
+        options = simpleData.options
+        parameters.push(options)
+    }
```

Comment:
I have to admit that I am very confused with `params`, `options` and `defaultOptions`. Can you explain what are their usage, please?

Reply:
args are all the arguments provided by the user. parameters are all the parameters needed by the function that we will call.
The parameters must have options as the last argument.

The trick is that sometimes we receive partial options (like {logs: true}) from the users or no options at all. But the functions always need options. So we check what we receive and we merge the defaultOptions (with all the options by default) with the received options to make sure the functions have everything they need.

It's also a way to avoid to provide all the options all the time when using the library.